provisioners/roles/ckan/tasks/main.yml

```diff
@@ -20,38 +19,74 @@
     group="{{ckan_group}}"
     mode=0755
     recurse=yes
-  sudo: yes
+  become: "{{use_sudo}}"
 
+- name: create apache server root
+  file:
+    path="{{apache_server_root}}"
+    state=directory
+    owner="{{ckan_user}}"
+    group="{{ckan_group}}"
+    mode=0755
+    recurse=yes
+  become: "{{use_sudo}}"
+
+- name: check if apache run dir exists
+  stat:
+    path: "{{apache_server_root}}/run"
+  register: apache_run_dir_stat
+
+- name: create apache run dir
+  file:
+    path="{{apache_server_root}}/run"
+    state=directory
+    owner="{{ckan_user}}"
+    group="{{ckan_group}}"
+    mode=0755
+    recurse=yes
+  when: not apache_run_dir_stat.stat.exists
+  become: "{{use_sudo}}"
+
+- name: create apache wsgi socket dir
+  file:
+    path="{{apache_server_root}}/sockets"
+    state=directory
+    owner="{{ckan_user}}"
+    group="{{ckan_group}}"
+    mode=0755
+    recurse=yes
+  become: "{{use_sudo}}"
+
+#Make the ckan venv
+- name: make the virtual environment
+  shell: "/usr/local/bin/virtualenv $(basename {{ckan_venv}})"
+  args:
+    creates: "{{ckan_venv}}"
+    chdir: "{{ckan_venv}}/.."
+
+#This version of python throws SNIMissingWarning unless an SNI extension to TLS is available
+- name: pip install ssl extensions
+  shell: "source {{ckan_venv}}/bin/activate && python -m pip install requests[security]"
+
+- name: pip install python-ldap
+  shell: "source {{ckan_venv}}/bin/activate && python -m pip install python-ldap"
+
+- name: pip install mod_wsgi
+  shell: "source {{ckan_venv}}/bin/activate && python -m pip install mod_wsgi"
+
+#Need to use python -m pip because when installing on mesos, you can't run pip directly due to the #! line being too long
 - name: pip install CKAN
-  pip: 
-    name="git+https://github.com/ckan/ckan.git@{{ckan_version_tag}}#egg=ckan"
-    virtualenv="{{ckan_venv}}"
-    virtualenv_site_packages=no
-    extra_args="-e"
-  environment: 
-    # Depending on the user running this command, we need
-    # to ensure /usr/local/bin is in the path so that 
-    # virtualenv is accessible.  
-    PATH: /usr/local/bin:/bin:/usr/bin
-  sudo: yes
+  shell: "source {{ckan_venv}}/bin/activate && python -m pip install -e git+https://github.com/ckan/ckan.git@{{ckan_version_tag}}#egg=ckan"
 
 - name: pip install CKAN requirements
-  pip:
-    requirements="{{ckan_venv}}/src/ckan/requirements.txt"
-    virtualenv="{{ckan_venv}}"
-  environment: 
-    # This is a bit of a hack so that we don't have to spend
-    # time gathering facts. We could just use anisble_env.PATH
-    PATH: /usr/pgsql-9.4/bin:/usr/local/bin:/bin:/usr/bin
-  sudo: yes
+  shell: "source {{ckan_venv}}/bin/activate && PATH=\"/usr/pgsql-9.4/bin:/usr/pgsql-9.5/bin:$PATH\" python -m pip install -r {{ckan_venv}}/src/ckan/requirements.txt"
```

Comment:
Same here, you could explicitly use the virtualenv's python, and also potentially use Ansible's "environment" option like here: https://github.com/cfpb/aurora/blob/2ad6bd89dec788cc787f9b4883e82d0bf25ff25c/deploy/roles/sqitch/tasks/main.yml#L53, but with ansible_env.PATH to retrieve the target's PATH variable.